### PR TITLE
feat (labs/analyzer): add support for mixins

### PR DESCRIPTION
**Pull request by 43081j**
Just a draft for now since it still needs tests, and the logic can probably be tidied up a bit (reduce some duplication).

### CODE REVIEW

**Comment by justinfagnani** [comment]
@kevinpschaaf had some work here too. @kevinpschaaf can you take a look and point @43081j to your branch?

File: packages/labs/analyzer/src/lib/javascript/classes.ts [hunk 1]
```diff
@@ -223,13 +223,61 @@ export const getHeritage = (
...
+  }
+
+  if (analyzer.typescript.isCallExpression(firstArg)) {
+    return getIdentifierFromMixin(firstArg, analyzer);
+  }
+
+  return null;
+};
+
+const getMixinsFromExpression = (
```
Comment (43081j) [inline packages/labs/analyzer/src/lib/javascript/classes.ts:247]: im thinking these two functions might be better as one which returns a `ClassHeritage` or something, so we traverse only once and get the mixins/superclass back in one go

**Comment by 43081j** [comment]
if one of you already has it, im happy to throw this branch away! just let me know

**Comment by justinfagnani** [comment]
I'm not sure the state of it, just wanted to raise it!

**Comment by 43081j** [comment]
Closing because #4147 will solve this

**Pull request closed without merging**